File: src/main/java/com/sbna/fxfeed/controller/MarketPriceController.java
package com.sbna.fxfeed.controller;

import com.sbna.fxfeed.repository.MarketPriceRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import static java.util.Optional.ofNullable;

@RestController
@RequestMapping("/prices")
@RequiredArgsConstructor
public class MarketPriceController {

    private final MarketPriceRepository marketPriceRepository;

    @GetMapping("/latest")
    @ResponseBody
    public String latestPrice(@RequestParam String name) {
        return ofNullable(marketPriceRepository.findLatestPrice(name.trim()))
                .map(it ->
                        String.format("""
                                ask: %s
                                bid: %s""", it.getAskPrice(), it.getBidPrice())).orElse("No result");

    }

}

File: src/main/kotlin/com/sbna/fxfeed/controller/MarketPriceController.kt
package com.sbna.fxfeed.controller

import com.sbna.fxfeed.repository.MarketPriceRepository
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/prices")
class MarketPriceController(private var marketPriceRepository: MarketPriceRepository) {

    @RequestMapping(
        value = ["/latest"], method = [RequestMethod.GET], produces = ["application/json"]
    )
    @ResponseBody
    fun latestPrice(@RequestParam name: String): String {
        val price = marketPriceRepository.findLatestPrice(name)

        price.also {
            return """
                |ask: ${it?.askPrice}
                |bid: ${it?.bidPrice}
                |""".trimMargin()
        }
    }

}